packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -118,28 +142,33 @@ class _Compiler {
         while (compilationQueue.isNotEmpty) {
           final _CompilationRequest request = compilationQueue.first;
           printTrace('Compiling ${request.path}');
+          compiler ??= createCompiler();
+          suppressOutput = false;
           final String outputPath = await compiler.recompile(request.path,
             <String>[request.path],
             outputPath: outputDill.path,
           );
-          // Copy output dill next to the source file.
-          final File kernelReadyToRun = await fs.file(outputPath).copy(
-              request.path + '.dill');
-          compiler.accept();
-          compiler.reset();
-          request.result.complete(kernelReadyToRun.path);
+
+          // Check if the compiler produced the output. It could have crashed
+          // in which case outputPath would be null. In this case pass
+          // null upwards to the consumer and shutdown the compiler.
+          if (outputPath == null) {
+            request.result.complete(null);
+            await shutdown();
```

Comment:
It's not clear why we are destroying compiler if there were compiler was not able to compile input source, was not able to produce kernel file.

Reply:
To avoid bizarre errors because compiler got into a weird state.

Reply:
Similar to handling of `quit` command, I would argue that if compiler gets into weird state in presence of compilation errors, that is valuable to know and should be filed as an error against compiler.